paasta_tools/long_running_service_tools.py

```diff
@@ -345,20 +360,47 @@ def limit_instance_count(self, instances: int) -> int:
         return max(self.get_min_instances(), min(self.get_max_instances(), instances))
 
     def get_autoscaling_params(self) -> AutoscalingParamsDict:
-        default_params: AutoscalingParamsDict = {
-            "metrics_provider": "cpu",
+        default_provider_params: MetricsProviderDict = {
+            "type": MetricsProviderType.CPU,
             "decision_policy": "proportional",
             "setpoint": DEFAULT_AUTOSCALING_SETPOINT,
         }
-        return deep_merge_dictionaries(
-            overrides=self.config_dict.get("autoscaling", AutoscalingParamsDict({})),
-            defaults=default_params,
+
+        params = copy.deepcopy(
+            self.config_dict.get("autoscaling", AutoscalingParamsDict({}))
         )
+        if "metrics_providers" not in params or len(params["metrics_providers"]) == 0:
+            params["metrics_providers"] = [default_provider_params]
+        else:
+            for i in range(0, len(params["metrics_providers"])):
+                params["metrics_providers"][i] = deep_merge_dictionaries(
+                    overrides=params["metrics_providers"][i],
+                    defaults=default_provider_params,
+                )
```

Comment:
non-blocking suggestion:
```suggestion
            params["metrics_providers"] = [
                deep_merge_dictionaries(
                    overrides=provider,
                    defaults=default_provider_params,
                )
                for provider in params["metrics_providers"]
            ]
```
i think this would do the same thing and get rid of the indexing - at the cost of folks needing to read a list-comp

(i'm personally somewhat averse to `for i in range(...)` and explicit indexing, but generally just for code-style reasons :p)

Reply:
Oh that's smart.  I don't know why I didn't do that.  I don't like explicit indexing either.